flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -293,7 +294,7 @@ private static void callMainMethod(Class<?> entryClass, String[] args) throws Pr
 		} catch (InvocationTargetException e) {
 			Throwable exceptionInMethod = e.getTargetException();
 			if (exceptionInMethod instanceof Error) {
-				throw (Error) exceptionInMethod;
+				throw (Error) ClusterEntryPointExceptionUtils.tryEnrichClusterEntryPointError(exceptionInMethod);
```

Comment:
I think the `PackagedProgram` is not the best place to call the enrichment method because this class can also be used on the client side when generating a `JobGraph` which is later submitted to the cluster. I think either the `JarRunHandler` and the `JarPlanHandler` or even the `AbstractHandler` might be a better place for the enrichment.